src/manager.spec.js

```diff
@@ -530,6 +530,8 @@ describe('AdManager', function() {
 
     it('- emits a dfpImpressionViewable event', function() {
       expect(eventSpy).to.have.been.called;
+      TestHelper.stub(adManager.adUnits.units.header, 'onImpressionViewable');
+      adManager.onImpressionViewable(event);
```

Comment:
This test should actually assert:

`expect(adManager.adUnits.units.header.onImpressionViewable.called).to.be.true;

And we should add similar test assertions for the other optional ad event callbacks you introduced.

Reply:
newest commits break out callbacks into separate assertions